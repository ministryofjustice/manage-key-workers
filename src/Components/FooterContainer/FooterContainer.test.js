import React from 'react'
import renderer from 'react-test-renderer'

import FooterContainer from '.'

const props = {
  supportUrl: '/',
  prisonStaffHubUrl: '/prisonStaffHubUrl/',
}

describe('<FooterContainer />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<FooterContainer {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render a link with the supportUrl prop', () => {
    const { root } = renderer.create(<FooterContainer {...props} />)
    const link = root.findByProps({ href: props.supportUrl })
    expect(link).toBeDefined()
    expect(link.props.children).toBe('Feedback and support')
  })

  it('should render a terms and conditions link with the prisonStaffHubUrl prop', () => {
    const { root } = renderer.create(<FooterContainer {...props} />)
    const link = root.findByProps({ href: `${props.prisonStaffHubUrl}content/terms-conditions` })
    expect(link).toBeDefined()
    expect(link.props.children).toBe('Terms and conditions')
  })
})
