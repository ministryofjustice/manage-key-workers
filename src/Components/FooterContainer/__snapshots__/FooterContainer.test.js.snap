// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterContainer /> should render correctly 1`] = `
<footer
  className="src__FooterContainer-sc-1t3c5e2-0 bHUhYt sc-dkzDqf koOAoh"
  role="contentinfo"
>
  <div
    className="width-container__WidthContainer-sc-213f5v-0 sc-gsnTZi ndzbb bgKgVb"
  >
    <div
      className="meta__Meta-b55tvf-0 kiiBfs"
    >
      <div
        className="meta-item__MetaItem-nseoos-0 jMDshW"
      >
        <span
          className="src__VisuallyHidden-nxwj55-0 gfekfu"
        >
          <h2
            className="heading__StyledHeading-sc-1dsobun-0 hkCaGQ"
            size="LARGE"
          >
            Support links
          </h2>
        </span>
        <ul
          className="unordered-list__UnorderedList-sc-1r7pzdm-0 dEFoQW"
        >
          <li
            className="list-item__ListItem-hn98ju-0 kdLuWH"
          >
            <a
              className="sc-bczRLJ eWiYH link__StyledFooterLink-sc-1vhdcoj-0 bTGwdP"
              href="/"
            >
              Feedback and support
            </a>
          </li>
          <li
            className="list-item__ListItem-hn98ju-0 kdLuWH"
          >
            <a
              className="sc-bczRLJ eWiYH link__StyledFooterLink-sc-1vhdcoj-0 bTGwdP"
              href="/prisonStaffHubUrl/content/terms-conditions"
              target="_blank"
            >
              Terms and conditions
            </a>
          </li>
        </ul>
        <div>
          <svg
            className="licence__LicenceLogo-sc-76bcjk-0 kYWaXG"
            focusable="false"
            height="17"
            role="presentation"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            className="licence__LicenceDescription-sc-76bcjk-1 dznhEZ"
          >
            All content is available under the
             
            <a
              className="sc-bczRLJ eWiYH link__StyledFooterLink-sc-1vhdcoj-0 bTGwdP"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
      </div>
      <div
        className="meta-item__MetaItem-nseoos-0 iGcKAb"
      >
        <a
          className="sc-bczRLJ copyright__StyledCopyright-sc-1i22iwf-0 eWiYH dEIuep"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          image={
            Object {
              "height": 102,
              "source": "govuk-crest-2x.png",
              "width": 125,
            }
          }
        >
          Â© 
          Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;
